### PR TITLE
Add Rococo chain-spec

### PR DESCRIPTION
**Pull request by cmichi**
Closes https://github.com/paritytech/canvas-node/issues/76.

@lovelaced This is the config which we also used for the Canvas Testnet, so the secret seeds are still the same.

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -116,32 +148,34 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
...
-					get_account_id_from_seed::<sr25519::Public>("Ferdie"),
-					get_account_id_from_seed::<sr25519::Public>("Alice//stash"),
-					get_account_id_from_seed::<sr25519::Public>("Bob//stash"),
-					get_account_id_from_seed::<sr25519::Public>("Charlie//stash"),
-					get_account_id_from_seed::<sr25519::Public>("Dave//stash"),
-					get_account_id_from_seed::<sr25519::Public>("Eve//stash"),
-					get_account_id_from_seed::<sr25519::Public>("Ferdie//stash"),
+					local_testnet_root(),
+					// AccountId of the Canvas Testnet faucet
+					hex!("18c64aa111a8a0e6e4eed41d6d906c7614d745e48be3cfc13b6128e1d51f4405").into(),
```
Comment (cmichi) [inline node/src/chain_spec.rs:157]: @lovelaced We want to use the Relaychain token for Canvas, do you know if the other parachains which do this just use the Rococo faucet then? We could remove this then.

File: node/src/chain_spec.rs [hunk 2]
```diff
@@ -116,32 +148,34 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
...
+					// AccountId of an account which `ink-waterfall` uses for automated testing
+					hex!("0e47e2344d523c3cc5c34394b0d58b9a4200e813a038e6c5a6163cc07d70b069").into(),
```
Comment (cmichi) [inline node/src/chain_spec.rs:159]: @riusricardo Do you know if it's even possible to have endowed accounts if we use the Relaychain currency?

Comment (riusricardo) [reply node/src/chain_spec.rs:159]: We can but we don't do it now (Statemint isn't connected to Rococo). The consequence of doing it is that we'll be minting extra native tokens from the relay chain on the parachain. However we can do it for Rococo and put a big warning/TODO so we don't forget changing it when we create the other runtimes.

Comment (riusricardo) [reply node/src/chain_spec.rs:159]: We can but we don't do it now (Statemint isn't connected to Rococo). The consequence of doing it is that we'll be minting extra native tokens from the relay chain on the parachain. However we can do it for Rococo and put a big warning/TODO so we don't forget changing it when we create the other runtimes.

**Comment by athei** [comment]
What is the reason to have the spec in code and as json? Why do we want this duplication that we need to keep in sync?

File: node/src/chain_spec.rs (hunk 1 above)
Comment (athei) [reply node/src/chain_spec.rs:157]: Token symbol is still set to "CAN" in this spec, though.

Comment (riusricardo) [reply node/src/chain_spec.rs:157]: @athei  I'm working on the currency related changes ;)

Comment (lovelaced) [reply node/src/chain_spec.rs:157]: @cmichi statemint on rococo (which doesn't exist at the moment) did in fact use ROC which you can just use the Rococo faucet for.

Comment (cmichi) [reply node/src/chain_spec.rs]: Okay, I've removed the faucet from the chain-spec!

**Comment by cmichi** [comment]
> What is the reason to have the spec in code and as json? Why do we want this duplication that we need to keep in sync? Honest question. There must be a reason but I don't know it.

My understanding was that this makes things easier for DevOps, but @lovelaced probably knows better?

**Comment by lovelaced** [comment]
> > What is the reason to have the spec in code and as json? Why do we want this duplication that we need to keep in sync? Honest question. There must be a reason but I don't know it.
> 
> My understanding was that this makes things easier for DevOps, but @lovelaced probably knows better?

We generally deploy the spec as a json file but as long as it's available via the `--chain` argument it doesn't really matter.

**Comment by cmichi** [comment]
>We generally deploy the spec as a json file but as long as it's available via the --chain argument it doesn't really matter.

Ok, I've removed it then. It can be generated via
```
cargo run -- build-spec --raw --chain canvas-rococo
```

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -111,14 +112,11 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
 
 	ChainSpec::from_genesis(
 		// Name
-		"Canvas Local",
+		"Canvas",
 		// ID
-		"canvas_local",
+		"canvas_rococo",
```
Comment (athei) [inline node/src/chain_spec.rs:117]:
```suggestion
		"canvas-rococo",
```

Consistency with the protocol ID.

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -140,12 +138,11 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
...
 		},
 		// Bootnodes
-		// These will be configured using the JSON files in the `res` foler
-		Vec::new(),
+		vec![],
 		// Telemetry
 		None,
 		// Protocol ID
-		Some("canvas-rococo"),
+		Some("canvas-local"),
```
Comment (athei) [inline node/src/chain_spec.rs:145]: What is this good for? Does it need to match with some stuff in the relay chain? The ID set above is `canvas_rococo`. Maybe we should be consistent here?

```suggestion
		Some("canvas-rococo"),
```

**Review by athei — COMMENTED** [review]
I am not sure about all those ids. If possible they should all match?

File: node/src/chain_spec.rs (hunk 4 above)
Comment (cmichi) [reply node/src/chain_spec.rs:145]: According to [the docs on the chain spec](https://substrate.dev/docs/en/knowledgebase/integrate/chain-spec):

>nodes will only add peers who use the same `protocolId`.

And from [the rustdoc of `ProtocolId`](https://substrate.dev/rustdocs/latest/sc_network/config/struct.ProtocolId.html):

>Name of a protocol, transmitted on the wire. Should be unique for each chain. 

So I think the distinction between local and Rococo still makes sense.

Comment (athei) [reply node/src/chain_spec.rs:145]: Ahh okay. I was looking at the wrong function.

File: node/src/chain_spec.rs [hunk 5]
```diff
@@ -156,6 +153,93 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
...
+	// Give your base currency a unit name and decimal places
+	let mut properties = sc_chain_spec::Properties::new();
+	properties.insert("tokenSymbol".into(), "CAN".into());
+	properties.insert("tokenDecimals".into(), 12.into());
+
+	ChainSpec::from_genesis(
+		// Name
+		"Canvas",
+		// ID
+		"canvas_rococo",
```
Comment (athei) [inline node/src/chain_spec.rs:197]:
```suggestion
		"canvas-rococo",
```

Consistency. Local and rococo use the same spec id. That is OK? I did not found any information regarding this field apart from that id is the "id".

Comment (cmichi) [reply node/src/chain_spec.rs:197]: I've distinguished them now.

Regarding the protocol id I just checked
* https://github.com/substrate-developer-hub/substrate-parachain-template/blob/master/node/src/chain_spec.rs
* https://github.com/paritytech/cumulus/blob/master/polkadot-parachains/src/chain_spec.rs

A protocol id is never set anywhere there, I've removed it from our chain spec as well now.

File: node/src/chain_spec.rs (hunk 4 above)
Comment (riusricardo) [reply node/src/chain_spec.rs]: @cmichi  Don't remove the protocol id. It is used to identify different networks.

File: node/src/chain_spec.rs (hunk 5 above)
Comment (riusricardo) [reply node/src/chain_spec.rs:197]: It is used on the networking layer. We should keep it.

Comment (cmichi) [reply node/src/chain_spec.rs:197]: I've added it back in after talking to Ricardo why they are missing in the code I linked above (seems to be a mistake there).

**Comment by cmichi** [comment]
Would be great if you could give it a final review @athei @riusricardo @lovelaced :crossed_fingers:.

I'd like to get this in so that we can review https://github.com/paritytech/canvas-node/pull/78 properly.

**Review by athei — APPROVED** [review]

**Review by lovelaced — APPROVED** [review]

**Review by riusricardo — APPROVED** [review]

**Pull request merged**